prover/witness_vector_generator/src/generator.rs

```diff
@@ -8,7 +8,7 @@ use anyhow::Context as _;
 use async_trait::async_trait;
 use tokio::{task::JoinHandle, time::sleep};
 use zksync_config::configs::FriWitnessVectorGeneratorConfig;
-use zksync_dal::{fri_prover_dal::types::GpuProverInstanceStatus, ConnectionPool};
+use zksync_dal::ConnectionPool;
```

Comment:
It feels clunky that `ConnectionPool` is not a part of `zksync_db_connection` crate and you cannot really use `zksync_prover_dal` without it.

I thought that the logic would be as follows:
- `zksync_db_connection` defines _all_ the machinery required to interact with the database, and only it.
- `zksync_dal` (optionally renamed to `zksync_sequencer_dal`) defines the sequencer DALs and only them.
- `zksync_prover_dal` defines the prover DALs and only them.

With segregations completed, it'd mean that crates in the core workspace would depend *only* on `zksync_dal` and crates in the prover workspace would depend *only* on `zksync_prover_dal`.

Reply:
that's true, but it will be done in the next PR, where we will separate ConnectionPool's, in this PR we would need a separate crates for both storage processor and connection pool, which will be avoided in the next one - so consider this one more as a preparation PR to make it more easy to review(although this one is big too)